Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -112,4 +112,145 @@ protected override void ProcessRecord()
+    {
+        private class ParameterSetNames
+        {
+            public const string ByObject = "ByObject";
+            public const string ByValues = "ByValue";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Table object that will be sent to the server to be inserted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        public Table InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to put the table in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        override public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Dataset that you would like to add to. This field takes strings, Datasets, and DatasetRefrences.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByValues)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset), Property = nameof(Apis.Bigquery.v2.Data.Dataset.DatasetReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetReference), Property = nameof(DatasetReference.DatasetId))]
+        public string Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The TableId must be unique within the Dataset and match the pattern [a-zA-Z0-9_]+.
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        [ValidateLength(1, 1024)]
+        public string TableId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// User-friendly name for the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Description of the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The lifetime of this table from the time of creation (in seconds).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidateRange(3600, (long.MaxValue / 1000))]
+        public long Expiration { get; set; }
```

Comment:
It is sufficient because we are not testing that BigQuery works correctly. That is someone else's job, We are testing that our API works correctly.